servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/MultiAddressUrlHttpClientTest.java

```diff
@@ -253,26 +257,52 @@ void requestWithAsteriskFormRequestTargetWithHostHeader() {
         assertThat(subscriber.awaitOnError(), is(instanceOf(MalformedURLException.class)));
     }
 
+    @Test
+    void exactRequestAuthorityIsRepresentedInHostHeader() throws Exception {
+        try (StreamingHttpClient client = HttpClients
+                .forMultiAddressUrl(ID).defaultHttpPort(serverPort).buildStreaming()) {
+            // Request without a port in the authority
+            StreamingHttpRequest request = client.get(format("http://%s/200", serverHost));
+            // the host header should exactly match the authority provided.
+            requestAndValidate(client, request, OK, "/200", serverHost);
```

Comment:
why does the client in this case does not use the `serverPort` used as a default in the host header? (compared to the next example where it's set explicitly)

Reply:
The `host` header should represent the exact authority (without user-info) that is part of the request target. Since the request target didn't specify a port neither should the host header.